File: code/spark-demo/src/main/scala/test/c3/Test2.scala
package test.c3

import org.apache.spark.rdd.RDD
import org.apache.spark.util.LongAccumulator

/**
 * @Author 0726
 * @ClassName Test2
 * @createTime 2021年11月03日 10:17
 */
object Test2 {
  def main(args: Array[String]): Unit = {
    import org.apache.spark.{SparkConf, SparkContext}
    //1.创建SparkConf并设置App名称
    val conf: SparkConf = new SparkConf().setMaster("local[4]").setAppName("test")
    //2.创建SparkContext，该对象是提交Spark App的入口
    val sc: SparkContext = new SparkContext(conf)

    val sum: LongAccumulator = sc.longAccumulator("sum")

    val rdd1: RDD[String] = sc.textFile("datas/wc.txt")

    val rdd2: RDD[(String, Int)] = rdd1.flatMap(_.split(" ")).map((_,1))

    var a = 0
    val r = rdd1.reduce((agg,curr)=>agg+curr)
//    rdd1.foreach(x=>sum.add(x))

    println(sum.value)
    println(r)

  }
}

File: code/0726project/homework/src/day04/Test2.java
package day04;

public class Test2 {
    public static void main(String [] args) {
        int sum = 0;

        for (int i = 2; i <= 100; i+=2) {
            sum += i;
        }
        System.out.println("sum ="+sum);
    }
}

// l: for (int i = 1; i <= 4; i+=1) {
//           if(i==3){
//              continue l;
//           }//此if循环表示跳过i=3时的累加
//           sum += i;
//           }
// System.out.println("sum ="+sum);//该循环sum的结果为1+2+4